src/main/java/com/microsoft/aad/adal4j/MexParser.java

```diff
@@ -58,6 +58,38 @@
     private final static String SOAP_TRANSPORT_XPATH = "soap12:binding/@transport";
     private final static String SOAP_HTTP_TRANSPORT_VALUE = "http://schemas.xmlsoap.org/soap/http";
 
+    static BindingPolicy getPolicyFromMexResponseForIntegrated(String mexResponse) throws Exception {
```

Comment:
It might be better to refactor this function since a lot of is code duplication with getWsTrustEndpointFromMexResponse finction